File: 数组/模拟/3066.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> pq = new PriorityQueue<>();
        for (int n : nums) pq.add((long) n);

        int ans=0;
        while (pq.size() >= 2 && pq.peek() < k) {
            long x = pq.poll();
            long y = pq.poll();
            long v = (long) (Math.min(x, y) * 2 +  Math.max(x, y));
            pq.offer(v);
            ans++;
        }

        return ans;
    }
}

File: 力扣/数组/模拟/3066.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<long long, vector<long long>, greater<long long>> pq;

        for (int n : nums) pq.push(n);

        int ans=0;

        while (pq.size() >= 2 && pq.top() < k) {
            long long x = pq.top(); pq.pop();
            long long y = pq.top(); pq.pop();
            long long v = min(x, y)*2 + max(x, y);
            pq.push(v);
            ans++;
        }

        return ans;
    }
};
